models/battleship/game.go

```diff
@@ -6,25 +6,57 @@ import (
 	"github.com/google/uuid"
 )
 
+// const (
+// 	GameValidLowerBound = 0
+// 	GameValidUpperBound = GameGridSize - 1
+// )
+
+const (
+	GameDifficultyEasy int = iota
+	GameDifficultyNormal
+	GameDifficultyHard
+)
+
 const (
-	GameValidLowerBound = 0
-	GameValidUpperBound = GameGridSize - 1
+	GridSizeEasy   int = 6
+	GridSizeNormal int = 7
+	GridSizeHard   int = 8
 )
 
 type Game struct {
-	isFinished bool
-	Uuid       string
-	HostPlayer *Player
-	JoinPlayer *Player
-	Players    map[string]*Player
+	isFinished      bool
+	Uuid            string
+	HostPlayer      *Player
+	JoinPlayer      *Player
+	Players         map[string]*Player
+	Difficulty      int
+	GridSize        int
+	ValidLowerBound int
+	ValidUpperBound int
 }
 
-func NewGame() *Game {
-	return &Game{
-		Uuid:       uuid.NewString()[:6],
-		isFinished: false,
-		Players:    make(map[string]*Player),
+func NewGame(difficulty int) Game {
+	game := Game{
+		Uuid:            uuid.NewString()[:6],
+		isFinished:      false,
+		Players:         make(map[string]*Player),
+		Difficulty:      difficulty,
+		ValidLowerBound: 0,
 	}
+
+	var gsize int
```

Comment:
nit: naming here could be better, like `newGridSize` or `gridSize`

Reply:
Fixed